### ISSUE TITLE
Version solving failed on platform gating

### ISSUE THREAD

**Issue by fizcris** [python-poetry/poetry#8920]
- **Poetry version**: Version: 1.7.1
- **Python version**: Python 3.12
- **OS version and name**: Windows 11
- **pyproject.toml**: 

```
[tool.poetry]
name = "myapp"
version = "2.0.0"
authors = ["fizcris"]
readme = "README.md"

[tool.poetry.dependencies]
python = "^3.12"
matplotlib = "^3.8.2"
only_mac_dep = { version = "*", platform = "darwin" }


[build-system]
requires = ["poetry-core"]
build-backend = "poetry.core.masonry.api"
```

- [X] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) stable Poetry version, installed using a [recommended method](https://python-poetry.org/docs/#installation).
- [X] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [X] I have consulted the [FAQ](https://python-poetry.org/docs/faq/) and [blog](https://python-poetry.org/blog/) for any relevant entries or release notes.
- [X] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option) and have included the output below.

## Issue
Pooetry fails to resolve dependencies that do not apply to the target platform rather than ignoring them:

Error for command `poetry install` in windows:
`Because myapp depends on only_mac_dep (*) which doesn't match any versions, version solving failed.`

**Comment by dimbleby** [comment]
please show your poetry.lock

**Comment by fizcris** [comment]
I do not agree with that statement, as if the package is only available for a platform then it won't be able to resolve, as it is the case, while the app could run without needed to resolve those platform specific packages.

**Comment by dimbleby** [comment]
I am afraid there is really nothing to agree or disagree with.

poetry produces cross-platform lockfiles, so at solving time it needs to be able to resolve the whole solution for all platforms.  This is working as designed.

please close

**Comment by fizcris** [comment]
Closed as per poor software design.
